### PR TITLE
Accessibility improvements

### PR DESCRIPTION
**Pull request by 2aces**
#Added
- Accessibility improvement: use of roles and other wai-aria attributes
- Accessibility improvement: focus management when baguetteBox popup is
open

References and acknowledgments in code

Closes https://github.com/feimosi/baguetteBox.js/issues/98

### CODE REVIEW

File: src/baguetteBox.scss [hunk 1]
```diff
@@ -123,6 +123,10 @@
     font: 1.6em sans-serif;
     transition: background-color .4s ease;
 
+    &:focus {
+        background-color: rgba(50,50,50,.9);
+    }
+
     &:hover {
```
Comment (feimosi) [inline src/baguetteBox.scss]: This could be placed in one statement like so:
```css
&:focus,
&:hover {
   ...
}
```

**Comment by feimosi** [comment]
Thanks for your contribution! Looks great, now I need some time to review it.
Just could you remove the version bump and build part? I want to take care of releases myself :)

File: build/js/baguetteBox.js [hunk 2]
```diff
@@ -321,6 +348,15 @@
         }, 50);
         if(options.onChange)
             options.onChange(currentIndex, imagesElements.length);
+        overlayStatus = true;
+        lastFocus = document.activeElement;
+        // change focus to next button if more than one image. Otherwise, focus on close button
+        if ( 1 < imagesElements.length ) {
```
Comment (XhmikosR) [inline build/js/baguetteBox.js]: Can you remove the spaces inside the brackets following the current style?

File: build/js/baguetteBox.js [hunk 3]
```diff
@@ -520,9 +562,14 @@
         loadImage(index - 1, function() { preloadPrev(index - 1); });
     }
 
-    function bind(element, event, callback) {
+    // modified to allow useCapture option. Needed for trapFocusHandler and
+    // to conform to wai-aria accessibility guidelines on modals/popup
+    function bind(element, event, callback, useCapture) {
+        if (useCapture === undefined) {
+              useCapture = false;
```
Comment (XhmikosR) [inline build/js/baguetteBox.js]: Too many spaces.

File: build/js/baguetteBox.js [hunk 4]
```diff
@@ -108,6 +116,15 @@
     var touchendHandler = function(event) {
         touchFlag = false;
     };
+    
+    //traps focus inside baguetteBox Popup when active
+    // https://www.nczonline.net/blog/2013/02/12/making-an-accessible-dialog-box/
+    var trapFocusHandler = function(event) {
+        if ( true === overlayStatus && !overlay.contains(event.target) ) {
```
Comment (XhmikosR) [inline build/js/baguetteBox.js]: I'd flip the true to the right and remove the spaces too.

File: build/js/baguetteBox.js (hunk 2 above)
Comment (XhmikosR) [inline build/js/baguetteBox.js]: And move the check to the right.

**Comment by 2aces** [comment]
Not sure i understood properly all notes, but... anyway, there's a new commit following the suggestions and requests. Hope this works now.

PS: This is my first public contribution and I am using Github desktop. Forgive any mistakes made by this noob. :-)

**Comment by XhmikosR** [comment]
The branch seems messed up to be honest. Just check the diff.

**Comment by XhmikosR** [comment]
@feimosi: you want me to cherry pick this and fix it?

**Comment by feimosi** [comment]
I think we should wait for @2aces to clean this up.

**Comment by XhmikosR** [comment]
I doubt he can, that is why I offered ;)

**Comment by feimosi** [comment]
Ok, let's wait a day or two, then feel free to do it yourself :)

**Comment by 2aces** [comment]
@feimosi sorry for the mistakes. In order do avoid further mistaker, what would you advise me to do it?


- revert to the commit before I made any changes AND then apply my changes following your style and notes or
- clean up my last commit
- leave be and don't mess anything else?

**Comment by XhmikosR** [comment]
@2aces:

```shell
git fetch upstream
git rebase -i upstream/master
# properly rebase your branch
```

**Comment by 2aces** [comment]
Sorry, @feimosi @XhmikosR I am facing a serious health issue on my shoulder and could not address these issues and my bad commit. It will take a few weeks of therapy for me getting back to code. Therefore:

1 - I welcome anyone cherry-picking it
2 - I had a few more ideas. What is the best course of action, discuss it here in the few moments i feel able to type or code it (properly) when my shoulder is healed?

**Comment by XhmikosR** [comment]
All right, I'll try to cherry pick this, clean it up and make a new PR.

**Comment by feimosi** [comment]
@2aces sure, I understand and no problem. Get well soon!

**Comment by feimosi** [comment]
@2aces I'll merge the current state of the accessibility improvements (see #112). When you'll be able to discuss your other ideas, please create a new issue.

**Pull request closed without merging**